src/main/kotlin/com/pubnub/api/PNConfiguration.kt

```diff
@@ -239,31 +239,18 @@ open class PNConfiguration(
 
     /**
      * When `true` the SDK will resend the last channel state that was set using [PubNub.setPresenceState]
-     * for the current [userId] with every automatic heartbeat.
+     * for the current [userId] with every automatic heartbeat (if [heartbeatInterval] is greater than 0)
+     * and subscribe reconnection.
      *
-     * Applies only if [heartbeatInterval] is greater than 0 and [enableEventEngine] is true.
+     * Applies only when [enableEventEngine] is true.
      *
      * Defaults to `true`.
      *
-     * Please note that `sendStateWithHeartbeat` doesn't apply to state that was set on channel groups.
+     * Please note that `maintainPresenceState` doesn't apply to state that was set on channel groups.
      * It is recommended to disable this option if you set state for channel groups using [PubNub.setPresenceState],
-     * otherwise that state may be overwritten on the next heartbeat with individual channel states.
+     * otherwise that state may be overwritten with individual channel states.
      */
-    var sendStateWithHeartbeat = true
-
-    /**
-     * When `true` the SDK will resend the last channel state that was set using [PubNub.setPresenceState]
-     * for the current [userId] when reconnecting subscriptions (i.e. subscribe calls with timetoken == 0).
-     *
-     * Applies only if [enableEventEngine] is true.
-     *
-     * Defaults to `true`.
-     *
-     * Please note that `sendStateWithSubscribe` doesn't apply to state that was set on channel groups.
-     * It is recommended to disable this option if you set state for channel groups using [PubNub.setPresenceState],
-     * otherwise that state may be overwritten on the next subscribe reconnection with individual channel states.
-     */
-    var sendStateWithSubscribe = true
+    var maintainPresenceState = true
```

Comment:
Is this somewhere documented that default is true ?

Going with default (maintainPresenceState = true) client may experience strange behaviour of SDK when setting state for channel groups. Are we ok with this?

Reply:
As far as I remember we already had this behavior in Kotlin in non-EE version, so that's why it's on by default

Reply:
I see.